src/lib/history.test.ts

```diff
@@ -0,0 +1,55 @@
+import { getMondayFromDate, getArticleViewCountForWeeks } from './history';
+
+const oneDay = 86400000;
+
+describe('getMondayFromDate', () => {
+    it('should return Monday midnight if today is Monday', () => {
+        const mondayMorning = new Date('2020-03-02T09:15:30');
+        const mondayMidnight = Math.floor(mondayMorning.getTime() / oneDay);
+
+        const mondayEvening = new Date('2020-03-02T19:25:00');
+        const got = getMondayFromDate(mondayEvening);
+        expect(got).toBe(mondayMidnight);
+    });
+
+    it('should return Monday midnight if today is some other day', () => {
+        const mondayMorning = new Date('2020-03-02T09:15:30');
+        const mondayMidnight = Math.floor(mondayMorning.getTime() / oneDay);
+
+        const fridayAfternoon = new Date('2020-03-06T16:35:00');
+        const got = getMondayFromDate(fridayAfternoon);
+        expect(got).toBe(mondayMidnight);
+    });
+});
+
+describe('getArticleViewCountForWeeks', () => {
+    it('should count views for one week properly', () => {
+        const history = [{ week: 18330, count: 45 }];
```

Comment:
What does the `week` number here represent? Weeks since the epoch?

Reply:
@tjmw It's the number of days since epoch - always a Monday at midnight so each key will increment 7 from the previous one.